packages/go_router/lib/src/delegate.dart

```diff
@@ -146,22 +146,38 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 
   bool _onPopPage(Route<Object?> route, Object? result) {
-    if (!route.didPop(result)) {
-      return false;
-    }
     final Page<Object?> page = route.settings as Page<Object?>;
     final RouteMatch? match = builder.getRouteMatchForPage(page);
     assert(match != null);
+    if (match!.route.onExit != null) {
+      // This function needs to return before calling onExit.
+      Future<void>.microtask(() async {
```

Comment:
why do you need a microtask?

Reply:
When testing showing a dialog in the `route.onExit`, if I don't wrap it with a microtask, this assertion will fail: https://github.com/flutter/flutter/blob/215fa3557b7e846e473d01fdad84bad584586e1f/packages/flutter/lib/src/widgets/navigator.dart#L4532

I think it's because we are trying to push a new route with `showDialog` while the current route is still being popped. Hence my comment above "This function needs to return before calling onExit", but obviously it's not clear enough.

Do you think it's a valid approach? Else, which comment would you have me write?

Reply:
use `scheduleMicrotask` from dart:async instead

Maybe update the comment to say
```
The `onExit` may perform update to the navigator, such as popping up a dialog. Using a microtask to ensure navigator finishes current update before calling the `onExit` 
```

Reply:
Done in [Use schedule microtask](https://github.com/flutter/packages/pull/3709/commits/906c687cf474a3bf04708f9427374082f9b726cb)